python/tests/test_account.py

```diff
@@ -859,27 +865,27 @@ def test_send_and_receive_message_markseen(self, acfactory, lp):
         ac2._evtracker.consume_events()
 
         lp.sec("sending a second message from ac1 to ac2")
-        msg_out2 = chat.send_text("message2")
+        msg3 = chat.send_text("message2")
 
         lp.sec("wait for ac2 to receive second message")
-        ev = ac2._evtracker.get_matching("DC_EVENT_INCOMING_MSG")
-        assert ev.data2 == msg_out2.id
-        msg_in2 = ac2.get_message_by_id(msg_out2.id)
+        msg4 = ac2._evtracker.wait_next_incoming_message()
 
         lp.sec("mark messages as seen on ac2, wait for changes on ac1")
-        ac2.mark_seen_messages([msg_in, msg_in2])
+        ac2.direct_imap.idle_start()
+        ac2.mark_seen_messages([msg2, msg4])
+        ac2.direct_imap.idle_check(terminate=True)
         lp.step("1")
         for i in range(2):
             ev = ac1._evtracker.get_matching("DC_EVENT_MSG_READ")
             assert ev.data1 > const.DC_CHAT_ID_LAST_SPECIAL
             assert ev.data2 > const.DC_MSG_ID_LAST_SPECIAL
         lp.step("2")
-        assert msg_out.is_out_mdn_received()
-        assert msg_out2.is_out_mdn_received()
+        assert msg1.is_out_mdn_received()
+        assert msg3.is_out_mdn_received()
 
-        lp.sec("check that a second call to mark_seen does not create change or smtp job")
+        lp.sec("try check that a second call to mark_seen doesn't happen")
```

Comment:
I found the old comment more understandable.

Reply:
will change back in another PR i am preparing.